infrastructure/edgeRedirectLambdaAssociations.ts

```diff
@@ -0,0 +1,124 @@
+import * as aws from "@pulumi/aws";
+import * as URLPattern from "url-pattern";
+import { CloudFrontRequestEvent } from "aws-lambda";
+import { LambdaEdge } from "./lambdaEdge";
+
+export function getEdgeRedirectAssociations():
+    aws.types.input.cloudfront.DistributionDefaultCacheBehaviorLambdaFunctionAssociation[] {
+    // https://aws.amazon.com/blogs/networking-and-content-delivery/handling-redirectsedge-part1/
+
+    const lambdaEdge = new LambdaEdge(
+        "redirects", {
+            func: (event: CloudFrontRequestEvent, context, callback) => {
+
+                // Get contents of response.
+                const request = event.Records[0].cf.request;
+
+                const redirect = getRedirect(request.uri);
+                if (redirect) {
```

Comment:
What do you think about flipping the conditions here so we can eliminate the need for an `else {}` block? I think it's more readable that way. That is,

```typescript
if (!redirect) {
  ....
}

// Do redirect stuff...
```

Reply:
Don't have a strong pref. I'll switch it up if you'd find that easier to read, sure.

Reply:
Yeah I find that large conditional blocks will only get larger and the little `else {}` will get pushed down even more and easy to miss.